pkgs/development/python-modules/fenics/default.nix

```diff
@@ -4,15 +4,15 @@
   fetchurl,
   fetchpatch,
   blas,
-  boost,
+  boost183,
   cmake,
   doxygen,
   eigen,
   hdf5,
   lapack,
   mpi,
   mpi4py,
-  numpy,
+  numpy_1,
```

Comment:
AIUI this isn’t allowed because Python package overrides have to be global (since versions of the same package can’t mix within one Python environment).

Reply:
Afair this was needed to fix the build, as fenics is broken on master.
```
=========================== short test summary info ============================
FAILED test/unit/misc/test_elements.py::Test_JIT::test_poisson - AttributeError: module 'numpy' has no attribute 'product'
```

This is not really related to the core change of this PR, so I can also drop it from this.
I simply fixed it because I wanted to make sure that all direct use-sites of boost in `python3Packages` work.

Reply:
Factored into https://github.com/NixOS/nixpkgs/pull/375132 now.